### PR TITLE
Update SLERT testing for 15-SP4

### PR DESCRIPTION
**Pull request by czerw**
- Related ticket: none
- Needles: none
- Verification run: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=5.19-dev&groupid=433

### CODE REVIEW

File: tests/kernel/install_ltp.pm [hunk 1]
```diff
@@ -143,11 +143,11 @@ sub install_build_dependencies {
       flex
       gcc
       git-core
-      kernel-default-devel
       libaio-devel
       libopenssl-devel
       make
     );
+    push @deps, is_rt ? 'kernel-rt-devel' : 'kernel-default-devel';
```
Comment (mdoucha) [inline tests/kernel/install_ltp.pm:150]: LGTM.

File: tests/rt/add_product.pm [hunk 2]
```diff
@@ -14,7 +14,7 @@ use testapi;
 sub run() {
     record_soft_failure 'poo#96158 - Adding RT product to control.xml';
     assert_screen 'startshell', 90;
-    assert_script_run 'sed -i \'/./{H;$!d} ; x ; s/\s*<\/base_product>\s*<\/base_products>/<\/base_product><base_product><display_name>SUSE Linux Enterprise Real Time 15 SP3<\/display_name><name>SLE_RT<\/name><version>15\.3<\/version><register_target>sle-15-\$arch<\/register_target><archs>x86_64<\/archs><\/base_product><\/base_products>/\' control.xml';
+    assert_script_run 'sed -i \'/./{H;$!d} ; x ; s/\s*<\/base_product>\s*<\/base_products>/<\/base_product><base_product><display_name>SUSE Linux Enterprise Real Time 15 SP4<\/display_name><name>SLE_RT<\/name><version>15\.4<\/version><register_target>sle-15-\$arch<\/register_target><archs>x86_64<\/archs><\/base_product><\/base_products>/\' control.xml';
```
Comment (pevik) [inline tests/rt/add_product.pm:17]: LGTM as is, but if you use `VERSION` variable (`s/-SP/ SP/` and `s/-SP/\\./`) it should work also for upcoming releases.

File: schedule/rt/online-migration-slertnext.yaml [hunk 3]
```diff
@@ -2,7 +2,6 @@
 description: 'Maintainer: thehejik; upgrade from SLERT12 SP3 to SLERT 12 SP4'
 name: 'online-migration-slert15spnext@64bit ( qemu )'
 schedule:
-  - installation/isosize
```
Comment (pevik) [inline schedule/rt/online-migration-slertnext.yaml:5]: "rt: Remove isosize from installation schedule": out of curiosity why? (might be worth to put the reason to the commit message).

**Review by pevik — APPROVED** [review]
LGTM, thanks for splitting into the comments which helps a lot to review.

File: data/autoyast_sle15/autoyast_rt_textmode.xml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      <!--listentry>
+        <media_url>http://download.suse.de/ibs/SUSE:/SLE-15-SP4:/Update:/Products:/SLERT/images/repo/SLE-15-SP4-Module-RT-POOL-x86_64-Media1/</media_url>
+        <alias>SLE_RT_OPENQA</alias>
+        <product>SLE_RT_TEST_OPENQA</product>
+        <product_dir>/</product_dir>
+        <priority config:type="integer">86</priority>
+        <name>Latest RT packages from IBS2</name>
+        <confirm_license config:type="boolean">false</confirm_license>
+        <ask_on_error config:type="boolean">false</ask_on_error>
+	  </listentry-->
```
Comment (b10n1k) [inline data/autoyast_sle15/autoyast_rt_textmode.xml:63]: Also this. Could you add a comment of why this is commented out?

File: data/autoyast_sle15/autoyast_rt_textmode.xml [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  </networking>
+  <users config:type="list">
+    <user>
+      <encrypted config:type="boolean">false</encrypted>
+      <fullname>Bernhard M. Wiedemann</fullname>
+      <gid>100</gid>
+      <home>/home/bernhard</home>
+      <password_settings>
+        <expire/>
+        <flag/>
```
Comment (b10n1k) [inline data/autoyast_sle15/autoyast_rt_textmode.xml:137]: Can those two be removed?

File: data/autoyast_sle15/autoyast_rt_textmode.xml [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        <inact>-1</inact>
+        <max>99999</max>
+        <min>0</min>
+        <warn>7</warn>
+      </password_settings>
+      <shell>/bin/bash</shell>
+      <uid>1000</uid>
+      <user_password>nots3cr3t</user_password>
```
Comment (b10n1k) [inline data/autoyast_sle15/autoyast_rt_textmode.xml:145]: better use `$6$vYbbuJ9WMriFxGHY$gQ7shLw9ZBsRcPgo6/8KmfDvQ/lCqxW8/WnMoLCoWGdHO6Touush1nhegYfdBbXRpsQuy/FTZZeg7gQL50IbA` hash

File: data/autoyast_sle15/autoyast_rt_textmode.xml [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        <expire/>
+        <flag/>
+        <inact/>
+        <max/>
+        <min/>
+        <warn/>
+      </password_settings>
+      <shell>/bin/bash</shell>
+      <uid>0</uid>
+      <user_password>nots3cr3t</user_password>
```
Comment (b10n1k) [inline data/autoyast_sle15/autoyast_rt_textmode.xml:163]: same for hash use

File: schedule/rt/prepare_migration.yaml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+description: 'Maintainer: kernel-qa@suse.de'
```
Comment (b10n1k) [inline schedule/rt/prepare_migration.yaml:1]: Maybe add some actual description?

File: schedule/rt/sle15/rt-product-textmode-autoyast.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+---
+name: autoyast_rt
+description: >
```
Comment (b10n1k) [inline schedule/rt/sle15/rt-product-textmode-autoyast.yaml:3]: Add context or remove?!

File: schedule/rt/sle15/rt-product-textmode-autoyast.yaml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+---
+name: autoyast_rt
```
Comment (b10n1k) [inline schedule/rt/sle15/rt-product-textmode-autoyast.yaml:2]: update? rt-product-textmode-autoyast or something? it is just too generic

File: tests/rt/prepare_migration.pm [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+# Copyright © 2022 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Prepare RT system for migration
+#            - do SCC cleanup
+#            - register RT system
```
Comment (b10n1k) [inline tests/rt/prepare_migration.pm:12]: I would rather have the reasoning which you described in the morning. For instance why do we need the cleanup. The module is small anyway and it is obvious what its steps are. But it is not clear why. wdyt?

File: data/autoyast_sle15/autoyast_rt_textmode.xml (hunk 6 above)
Comment (czerw) [reply data/autoyast_sle15/autoyast_rt_textmode.xml:145]: why?

Comment (b10n1k) [reply data/autoyast_sle15/autoyast_rt_textmode.xml:145]: because the password is a plain text. it will kinda prevent the exposure on git

Comment (czerw) [reply data/autoyast_sle15/autoyast_rt_textmode.xml:145]: You mean to don't expose password in? That password which is defined in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/09fc0113850caae8a838e1e7cfb77f31fceb21bf/lib/main_common.pm#L160 , really?

File: data/autoyast_sle15/autoyast_rt_textmode.xml (hunk 7 above)
Comment (czerw) [reply data/autoyast_sle15/autoyast_rt_textmode.xml:163]: Just no.

**Comment by czerw** [comment]
> i added some comments/questions from my side.

Thanks a lot for your comments, will try to address almost all of them.

File: schedule/rt/sle15/rt-product-textmode-autoyast.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
+name: autoyast_rt_textmode
```
Comment (b10n1k) [inline schedule/rt/sle15/rt-product-textmode-autoyast.yaml:1]: i would excepted to find `rt-product-textmode-autoyast` as name convention from  `rt-product-gnome-autoyast`. would you mind to change it?

**Review by b10n1k — COMMENTED** [review]
Could we rename also `schedule/rt/sle15/rt-product.yaml` to schedule/rt/sle15/rt-product-gnome.yaml. if you think it s too much we can ignore it for now and discuss it later

**Comment by czerw** [comment]
> Could we rename also `schedule/rt/sle15/rt-product.yaml` to schedule/rt/sle15/rt-product-gnome.yaml. if you think it s too much we can ignore it for now and discuss it later

I would keep it for next round, as we have already did many changes. Now i need to wait for our infra to fix scc proxy to have verification run.

File: tests/rt/add_product.pm (hunk 2 above)
Comment (czerw) [reply tests/rt/add_product.pm:17]: Rather not touch now, but i will include it in next improvements in August.

File: schedule/rt/online-migration-slertnext.yaml (hunk 3 above)
Comment (czerw) [reply schedule/rt/online-migration-slertnext.yaml:5]: Commit msg is now: We don't need to do isosize check for Online/Full mediums. It was mainly used to check that
iso will fit to CD/DVD, which is not relevent now.  Is it OK?

**Review by b10n1k — APPROVED** [review]
there are some minor things left but we can address them in another round. good to go from my side too.

**Pull request merged**